data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/VariableExpander.java

```diff
@@ -33,26 +42,46 @@ public VariableExpander(
+        patternPluginConfigValueTranslatorMap = new HashMap<>();
+        pluginConfigValueTranslators.forEach(pluginConfigValueTranslator -> {
+            if (!DEFAULT_DEPRECATED_PREFIX.equals(
+                    pluginConfigValueTranslator.getDeprecatedPrefix())) {
+                patternPluginConfigValueTranslatorMap.put(
+                        Pattern.compile(
+                                String.format(SECRETS_REFERENCE_PATTERN_STRING,
+                                        pluginConfigValueTranslator.getDeprecatedPrefix(), VALUE_REFERENCE_KEY)),
+                        pluginConfigValueTranslator);
+            }
+            patternPluginConfigValueTranslatorMap.put(
+                    Pattern.compile(
+                            String.format(SECRETS_REFERENCE_PATTERN_STRING,
+                                    pluginConfigValueTranslator.getPrefix(), VALUE_REFERENCE_KEY)),
+                    pluginConfigValueTranslator);
+        });
     }
 
     public <T> T translate(final JsonParser jsonParser, final Class<T> destinationType) throws IOException {
         if (JsonToken.VALUE_STRING.equals(jsonParser.currentToken())) {
             final String rawValue = jsonParser.getValueAsString();
-            return patternPluginConfigValueTranslatorMap.entrySet().stream()
-                    .map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey().matcher(rawValue), entry.getValue()))
-                    .filter(entry -> entry.getKey().matches())
-                    .map(entry -> {
-                        final String valueReferenceKey = entry.getKey().group(VALUE_REFERENCE_KEY);
-                        return objectMapper.convertValue(
-                                entry.getValue().translate(valueReferenceKey), destinationType);
-                    })
-                    .findFirst()
-                    .orElseGet(() -> objectMapper.convertValue(rawValue, destinationType));
+            final Matcher referenceMatcher = REFERENCE_PATTERN.matcher(rawValue);
+            if (referenceMatcher.matches()) {
+                LOG.debug("Searching for PluginConfigValueTranslator to match prefix on the raw value: {}", rawValue);
+                return patternPluginConfigValueTranslatorMap.entrySet().stream()
+                        .map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey().matcher(rawValue), entry.getValue()))
+                        .filter(entry -> entry.getKey().matches())
+                        .map(entry -> {
+                            final String valueReferenceKey = entry.getKey().group(VALUE_REFERENCE_KEY);
+                            return objectMapper.convertValue(
+                                    entry.getValue().translate(valueReferenceKey), destinationType);
+                        })
+                        .findFirst()
+                        .orElseGet(() -> {
+                            LOG.debug("No PluginConfigValueTranslator matches prefix on the raw value: {}, " +
```

Comment:
I think this should be a WARN level. If this happens, I'd guess most of the time it is a user error.

I'm also doubtful about the value of this behavior. I'd think that `\${{..}}` would be the right way to express this.

Reply:
Discussed offline. We will keep this behavior for backward compatibility and introduce skip on `${{}}` in the next major version.